### DISCUSSION TITLE
Easiest Way to POST from External Service?

### DISCUSSION THREAD

**Discussion by HenryHuevo** [pocketbase/pocketbase#3413] [Q&A] [answered]
I am trying to implement a feature where a php form submission from Site A POSTs to Site B, which is running pocketbase.  I want the POST request to be authenticated, but I am unsure of the best/easiest way of doing this.  For example, do I make a POST request to: `/api/collections/users/auth-with-password` with the body params of `identity` and `password`, grab the token from the request, then use it to make another request to `/api/collections/collectionIdOrName/records`, or would I just make a request to `/api/collections/collectionIdOrName/records` with body parameters of `username`, `password`, and `passwordConfirm`?

**Comment by HenryHuevo** [comment]
Nevermind, I simply applied more effort and figured it out.  Passing those body parameters solves this problem.  Self closing

**Comment by HenryHuevo** [comment]
Re-opening because I recognized that my API rule was to allow everyone to create, so of course the POST request succeeded.  Still looking for the api restriction rule to allow this.  I tried passing creds of an application user via basic auth, no luck, and I also tried passing the username/password through the BODY, which TECHNICALLY works when I specify a rule like `@request.data.username = "username_here"`, but I'm not sure if this should be handled a different way?

**Reply by ganigeorgiev** [reply]
I don't understand what do you mean by _" api restriction rule to allow this"_.

If you want to submit a new record to auth protected collection you'll need to authenticate first and then send the create request with `Authorization: TOKEN` header.

What is purpose of `@request.data.username = "username_here"`? Note that this doesn't perform authentication and just checks whether the submitted username matches "username_here".

**Reply by HenryHuevo** [reply]
Sorry for the lack of clarification.  I was looking to pass authentication in a single request along with the form submission data, but if I need to make an auth request to get the token, then a second request w/ the token and form data, then that is exactly what I will do.

`What is purpose of @request.data.username = "username_here"? Note that this doesn't perform authentication and just checks whether the submitted username matches "username_here".`

Yeah, that was me fumbling around in an attempt to use a "key" to permit the create record all in one request, but I think I am going to go the two request route if I need to.

The only issue I see with authing then passing the token is that I would only want this particular token to work for the submission of data and then essentially be useless/expire near-immediately.  I would change the "Auth Token" value in the PB admin dashboard, but that would effect ALL Auth tokens and would be undesirable.  Is there a way to handle this in PB?

**Reply by ganigeorgiev** [reply, marked as answer]
>  I was looking to pass authentication in a single request along with the form submission data

You can't unless you register a custom route (see [Use as framework](https://pocketbase.io/docs/use-as-framework/)).

> I would only want this particular token to work for the submission of data and then essentially be useless/expire near-immediately. 

But why? What exactly are your concerns? I'm just failing to understand your use case and what are you trying to do. If this token is requested on the PHP side you can safely discard it after your record create request (the tokens are not stored anywhere).

**Reply by HenryHuevo** [reply]
I guess my concern was "Hey, I don't want this token to be re-used", but you are correct in that it doesn't matter because after it gets called, I am not storing the token in my hook or my pb instance, and it gets discarded into the ether.

Forgive me for not being clear, I am new to webdev, but I think between all these posts I better understand my OWN use case, and how I will go about implementing pocketbase into the solution.  Thanks for the help.
